exporter_test.go

```diff
@@ -1299,6 +1307,17 @@ version = "4.5.6"
 	})
 }
 
+func checkEntrypoint(t *testing.T, image *fakes.Image, expectedDir string, expectedFile string) {
```

Comment:
It might be nice to pass the expected entrypoint into the function, then this could read like `assertHasEntrypoint(t, image, expectedEntrypoint)`. There'd be slightly more duplication in passing the correct expectation for Linux vs. Windows, but it might read easier.

Reply:
I don't mind to rename the function to `assertHasEntrypoint` but I think it's better not to pass the 2 entrypoints (one for each OS).
I think it's better to hide the differences between the entrypoints of linux and windows because it prevents code duplication and therefore typos while writing new tests.
In order to make the code clearer, maybe we can pass only one entrypoint and do string manipulations on it in the function in order to check the other expected entrypoint, although it might be confusing.
What do you think?

Reply:
Hmm... what about passing e.g., `filepath.Join(rootDir, "cnb", "process", "some-process-type"+execExt)` with some os-based constants like here: https://github.com/buildpacks/lifecycle/blob/fae1fe27c0acdfabe9538e3607e72af61c62ede9/cmd/flags.go#L17

Reply:
Here's an example of tests where we have os-based duplication: https://github.com/buildpacks/lifecycle/blob/main/archive/reader_test.go

I'd argue that the tests are nevertheless more readable this way...

Reply:
In general I agree with you that duplication is bad but I think we should also prioritize readability. It's nice when scanning through the tests if you can immediately understand what they are asserting and don't need to navigate elsewhere in the code to see what a helper function is doing. Though I'm already familiar with this code, I found it a bit confusing so I feel that others might as well...

I feel that https://github.com/buildpacks/lifecycle/pull/520#discussion_r568744924 could go a long way toward reducing the os-based duplication though. We'd have duplication of `filepath.Join` but all other logic could remain in the helper.

Reply:
I changed this code [here](https://github.com/buildpacks/lifecycle/pull/520/commits/972cf6de0cba7f026367a6dd49532bbf94ded7bf).